### ISSUE TITLE
some pb with intero

### ISSUE THREAD

**Issue by nrolland** [jakubfijalkowski/hlibsass#6]
The default configuration has `intero` say 

```
GHC runtime linker: fatal error: I found a duplicate definition for symbol
   __ZN4Sass7EmitterD0Ev
whilst processing object file
   /Users/nrolland/.stack/snapshots/x86_64-osx/nightly-2017-01-14/8.0.1/lib/x86_64-osx-ghc-8.0.1/hlibsass-0.1.5.0-DAuvZ5FMWJ2GN3MaQufO80/libsass.a
The symbol was previously defined in
   /Users/nrolland/.stack/snapshots/x86_64-osx/nightly-2017-01-14/8.0.1/lib/x86_64-osx-ghc-8.0.1/hlibsass-0.1.5.0-DAuvZ5FMWJ2GN3MaQufO80/libsass.a(emitter.o)
This could be caused by:
   * Loading two different object files which export the same symbol
   * Specifying the same object file twice on the GHCi command line
   * An incorrect `package.conf' entry, causing some object to be
     loaded twice.
intero: Could not on-demand load symbol '__ZN4Sass14Operation_CRTPIvNS_7InspectEEclEPNS_12Custom_ErrorE'

intero: 
lookupSymbol failed in relocateSection (RELOC_GOT)
/Users/nrolland/.stack/snapshots/x86_64-osx/nightly-2017-01-14/8.0.1/lib/x86_64-osx-ghc-8.0.1/hlibsass-0.1.5.0-DAuvZ5FMWJ2GN3MaQufO80/libsass.a: unknown symbol `___dso_handle'
intero: Could not on-demand load symbol '__ZN4Sass14Memory_Manager3addEPNS_13Memory_ObjectE'

intero: 
lookupSymbol failed in relocateSection (relocate external)
/Users/nrolland/.stack/snapshots/x86_64-osx/nightly-2017-01-14/8.0.1/lib/x86_64-osx-ghc-8.0.1/hlibsass-0.1.5.0-DAuvZ5FMWJ2GN3MaQufO80/libsass.a: unknown symbol `__ZN4Sass14Memory_Manager3addEPNS_13Memory_ObjectE'
intero: Could not on-demand load symbol '__ZN4Sass6Parser3lexIXadL_ZNS_8Prelexer10identifierEPKcEEEES4_bb'

```

Using an external libsass involves installing it globally `brew install libsass` and adding to the `stack.yaml`:
 
```
extra-deps:
- hlibsass-0.1.5.0

# Override default flag values for local packages and extra-deps
flags:
      hlibsass:
        externalLibsass: true  
```

Now intero should work.

It might be useful to add it to the readme.

**Comment by nrolland** [comment]
just upgraded to sierra and now this solution does not work. 
even `stack build`  was broken, but removing the flag did work.
no intero though, don't know what's going on.

edit : after upgrading to 8.0.2, and *not* adding the flag in `stack.yaml` everything works fine
edit : after upgrading stack it does not

```


hsass-0.4.0: build
Progress: 2/4
--  While building package hsass-0.4.0 using:
      /Users/nrolland/.stack/setup-exe-cache/x86_64-osx/Cabal-simple_mPHDZzAJ_1.24.2.0_ghc-8.0.2 --builddir=.stack-work/dist/x86_64-osx/Cabal-1.24.2.0 build --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: /Users/nrolland/Sync/clones/xquant2/.stack-work/logs/hsass-0.4.0.log

    Configuring hsass-0.4.0...
    Building hsass-0.4.0...
    Preprocessing library hsass-0.4.0...
    [ 1 of 11] Compiling Text.Sass.Values ( Text/Sass/Values.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Text/Sass/Values.o )
    [ 2 of 11] Compiling Text.Sass.Utils  ( Text/Sass/Utils.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Text/Sass/Utils.o )
    [ 3 of 11] Compiling Text.Sass.Values.Internal ( Text/Sass/Values/Internal.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Text/Sass/Values/Internal.o )
    [ 4 of 11] Compiling Text.Sass.Values.Utils ( Text/Sass/Values/Utils.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Text/Sass/Values/Utils.o )
    [ 5 of 11] Compiling Text.Sass.Functions ( Text/Sass/Functions.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Text/Sass/Functions.o )
    [ 6 of 11] Compiling Text.Sass.Functions.Internal ( Text/Sass/Functions/Internal.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Text/Sass/Functions/Internal.o )
    [ 7 of 11] Compiling Text.Sass.Options ( Text/Sass/Options.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Text/Sass/Options.o )
    [ 8 of 11] Compiling Text.Sass.Options.Internal ( Text/Sass/Options/Internal.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Text/Sass/Options/Internal.o )
    [ 9 of 11] Compiling Text.Sass.Internal ( Text/Sass/Internal.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Text/Sass/Internal.o )
    [10 of 11] Compiling Text.Sass.Compilation ( Text/Sass/Compilation.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Text/Sass/Compilation.o )
    [11 of 11] Compiling Text.Sass        ( Text/Sass.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Text/Sass.o )
    ld: warning: directory not found for option '-L/private/var/folders/9d/vgq7v77x27xb7v93gg85bcmc0000gn/T/stack79876/hlibsass-0.1.5.0/libsass/lib'
    ld: library not found for -lsass
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    `gcc' failed in phase `Linker'. (Exit code: 1)
```

**Comment by jakubfijalkowski** [comment]
Let's move the discussion to the jakubfijalkowski/hsass#4, cause these are exactly the same problems and there is more info about them (and I am unable to reliably track both issues).

**Comment by jakubfijalkowski** [comment]
Okay, I've tested this on macOS, and GHCi still does not work. I have no idea what should be done to make this work. I suspect that it is related to the fact that LibSass uses GCC to compile and default compiler for macOS is LLVM (although it seems that `gcc` is only an alias for LLVM on this system).

Currently, the only option is to use shared (tested, works) or external (tested) version of LibSass. I've updated the README with the fix, so I consider this issue resolved.
